testutils/mock_server/server.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+const http = require('http')
+
+const port = 3000
+
+const requestHandler = async (request, response) => {
+    const { headers, method, url } = request;
+
+    if (url == '/get_request' && method == 'GET') {
```

Comment:
This looks good and simple.
But there are a couple of scenarios that will not work like this:
- Response latency and multiple retries simulation (5xx error use-case)
- Different error scenarios, e.g. token expired, unknown partition
- Error checks (how can we check that the server got what it should and only that?)

Reply:
This change is a baseline